com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricCloudsLighting.cs

```diff
@@ -0,0 +1,71 @@
+namespace UnityEngine.Rendering.HighDefinition
+{
+    public partial class HDRenderPipeline
+    {
+        // Ref: "Efficient Evaluation of Irradiance Environment Maps" from ShaderX 2
+        Vector3 SHEvalLinearL0L1(Vector3 N, Vector4 shAr, Vector4 shAg, Vector4 shAb)
+        {
+            Vector4 vA = new Vector4(N.x, N.y, N.z, 1.0f);
+
+            Vector3 x1;
+            // Linear (L1) + constant (L0) polynomial terms
+            x1.x = Vector4.Dot(shAr, vA);
+            x1.y = Vector4.Dot(shAg, vA);
+            x1.z = Vector4.Dot(shAb, vA);
+
+            return x1;
+        }
+
+        Vector3 SHEvalLinearL2(Vector3 N, Vector4 shBr, Vector4 shBg, Vector4 shBb, Vector4 shC)
+        {
```

Comment:
Do'nt we already have this code somewhere? 
We should not hide generic function like this in file like volumetrics, should be in spherical harmonics. cs file in Core or LightUtil.cs or similar

Reply:
Pretty sure we have this kind of function for PBS sky already

Reply:
This PR just moves it inside the clouds code, didn't add it in this PR